app/controllers/users_controller.rb

```diff
@@ -45,41 +47,14 @@ def user_params
     params.permit(:name, :address, :city, :state, :zip, :email, :password, :password_confirmation)
   end
 
-  def e_params
-    params.permit(:name, :address, :city, :state, :zip, :email)
-  end
-
-  def password_params
-    params.permit(:password, :password_confirmation)
-  end
-
-  def update_e_params(bool = false)
-    if bool == true
-      @user = @user.update(e_params)
-      flash[:success] = "Your profile has been updated."
-      redirect_to '/profile'
-    else
-      flash[:error] = "This email is already used."
-      render :edit
-    end
-  end
-
```

Comment:
Cleaned a lot of this up in light of Meg's feedback in the last review. All tests still passing

Reply:
Wow! That's a lot of cleanup! Well done!